### DISCUSSION TITLE
How to write test with supertest

### DISCUSSION THREAD

**Discussion by 7elven** [kartikk221/hyper-express#96] [Q&A] [answered]
I used [supertest](https://github.com/visionmedia/supertest) to write api test with express project
Now, I want to change from `express` to `hyper-express` but test doesn't work

##### app.ts
```js
import express from 'express'
const app = express()
app.use(express.json({ limit: '50mb' }))
app.use(express.urlencoded({ extended: false }))

export default app
```

##### app.ts
```js
import HyperExpress from 'hyper-express'

const app = new HyperExpress.Server({ max_body_length: 50 * 1000000 }) // 50mb

export default app
```

---
##### router.test.ts
```js
import request from 'supertest'

import app from '../../../app'

describe('GET /', () => {
  it('Should response status code 200', async () => {
    const res = await request(app).get('/')
    expect(res.status).toEqual(200)
  })
})

afterAll((done) => {
  done()
})
```

This test is work on `express` but on `hyper-express` is show error 
```
TypeError: app.address is not a function
```

How to fix this? or How Hyper-Express write api test?

**Comment by kartikk221** [comment, marked as answer]
HyperExpress is not 100% drop-in compatible with ExpressJS and it seems that some of the logic is trying to call a method called `Server.address()` which currently is not implemented with HyperExpress.

You can add this method yourself onto your `app` if you understand the behavior of this method or wait for future updates which may add compatibility for this method.
